docs/references/architecture/adr-113-mempool-lanes.md

```diff
@@ -3,43 +3,134 @@
+they are not discriminated as to which are disseminated first or which transactions the mempool
+offers to the proposer when creating the next block. In practice, however, not all transactions have
+the same importance for the application logic, especially when it comes to latency requirements.
+Depending on the application, we may think on countless categories of transactions based on their
+importance and requirements, spanning from IBC messages, transactions for exchanges, for smart
+contract execution, for smart contract deployment, grouped by SDK modules, and so on. Even
+transactions prioritized by economic incentives could be given a preferential treatment. Or big
+transactions, regardless of their nature, could be categorized as low priority, to mitigate
+potential attacks on the mempool.
+
+The goal of this document is thus to propose a mechanism enabling the mempool to prioritize
+transactions by *classes*, for processing and dissemination, directly impacting block creation. In
+IP networking terminology, this is known as Quality of Service (QoS). By providing certain QoS
+guarantees, developers will be able to more easily estimate when transactions will be processed and
+included in a block. 
+
+In practical terms, we envision an implementation of the transaction class abstraction as *mempool
+lanes*. The application will be allowed to split the mempool transaction space into a hierarchy of
+lanes, with each lane operating as an independent mempool. At the same time, all of them need to be
+coordinated to ensure the delivery of the desired levels of QoS.
+
+Note that improving the dissemination protocol to reduce bandwidth and/or latency is a separate
+concern and falls outside the scope of this proposal.
+
+## Properties
+
+Before jumping into the design of the proposal, we define more formally the properties supported by
+the current implementation of the mempool. Then we state what properties the new mempool should
+offer to guarantee the desired QoS.
+
+The following definition is common to all properties. 
+
+:memo: _Definition_: Given any two different transactions `tx1` and `tx2`, we say that `tx1` is
+*processed and disseminated before* `tx2`, when:
+- `tx1` is reaped from the mempool to form a block proposal before `tx2`,
+- `tx1` is rechecked before `tx2`, and
+- `tx1` is disseminated to a peer before `tx2`.
+
+Note that in the current implementation there is one dissemination routine per peer, so it could
+happen that `tx2` is sent to a peer before `tx1` is sent to a different peer.
+
+### Current mempool
+
+As stated above, the current mempool offers a best-effort FIFO ordering of transactions. We state
+this property as follows.
+
+:parking: _Property_ **FIFO ordering of transactions**: We say that the mempool makes a best effort
+in maintaining the FIFO ordering of transactions when transactions are validated, processed, and
```

Comment:
When you say "processed" in this sentence, what do you have in mind? Added to the mempool? Run CheckTX on it?

Reply:
"processed and disseminated" as defined in the previous definition. I clarified in the text.